cli/lsp/documents.rs

```diff
@@ -379,14 +393,20 @@ impl Document {
     } else {
       None
     };
-    let source = SourceTextInfo::new(content);
+    let dependencies = if let Some(Ok(module)) = &maybe_module {
+      Arc::new(module.dependencies.clone())
+    } else {
+      self.0.dependencies.clone()
```

Comment:
This is what fixes the issue, we carry over the dependencies when there is an invalid parse.  This allows completions/suggestions/etc to still be processed by tsc when the module contains an unparsable syntax error.